x-pack/legacy/plugins/beats_management/server/lib/beats.ts

```diff
@@ -249,3 +248,12 @@ function addToResultsToResponse(key: string, response: any, assignmentResults: a
   });
   return response;
 }
+
+export function findNonExistentItems(items: Array<{ id: string }>, requestedItems: string[]) {
+  return requestedItems.reduce((nonExistentItems: string[], requestedItem: string, idx: number) => {
```

Comment:
Only util that was actually not dead code. Moved it to the only file importing it.

Reply:
is it a sophisticated version of filter?! 
```js
return requestedItems.filter((r) => items.every((item) => item.id !== r));
```
requestedItems seems to be `string[]`

Reply:
An over engineered version of filter yea. However due to the test coverage of the plugin, I'm trying to avoid changing anything that is not necessary.